beetsplug/lastgenre/__init__.py

```diff
@@ -158,6 +159,27 @@ def sources(self):
         elif source == 'artist':
             return 'artist',
 
+    def _get_depth(self, tag):
+        """Find the depth of a tag in the genres tree.
+        """
+        depth = None
+        for key, value in enumerate(self.c14n_branches):
+            if tag in value:
+                depth = value.index(tag)
+                break
+        return depth
+
+    def _sort_by_depth(self, tags):
+        """Given a list of tags, sort the tags by their depths in the
+        genre tree.
+        """
+        tags_by_depth = []
+        for key, value in enumerate(tags):
+            depth = self._get_depth(value)
+            tags_by_depth.append({'tag': value, 'depth': depth})
+        tags_by_depth = sorted(tags_by_depth, key=lambda k: k['depth'], reverse=True)
+        return [i['tag'] for i in tags_by_depth]
```

Comment:
One way to do this sort that might be slightly shorter to write would be to use tuples. For example, something like this:

```py
depth_tag_pairs = [self._get_depth(t), t for t in tags]
depth_tag_pairs.sort(reverse=True)
return [p[0] for p in depth_tag_pairs]
```

That will automatically sort by the first element in the tuple first.